src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid

```diff
@@ -284,28 +284,44 @@
 {%         elseif operation.HasSuccessResponse -%}
                     {
                         var responseData_ = response_.Content == null ? null : await response_.Content.ReadAsStringAsync().ConfigureAwait(false); 
-                        throw new {{ ExceptionClass }}("{{ operation.DefaultResponse.ExceptionDescription }}", (int)response_.StatusCode, responseData_, headers_, null);
+                        throw new {{ ExceptionClass }}("{{ operation.DefaultResponse.ExceptionDescription }}", status_, responseData_, headers_, null);
                     }
-{%         endif -%}
-{%     else -%}
-                    if (status_ != "200" && status_ != "204")
-                    {
-                        var responseData_ = response_.Content == null ? null : await response_.Content.ReadAsStringAsync().ConfigureAwait(false); 
-                        throw new {{ ExceptionClass }}("The HTTP status code of the response was not expected (" + (int)response_.StatusCode + ").", (int)response_.StatusCode, responseData_, headers_, null);
-                    }
-{%     endif -%}
-{%     if operation.HasDefaultResponse == false or (operation.DefaultResponse.HasType == false and operation.HasSuccessResponse == false) -%}
-{%         if operation.HasResultType -%}
-        
+{%        elseif operation.HasResultType -%}
```

Comment:
I admit that this is quite hard to review with the diff tools.
I took the part inside `if operation.HasDefaultResponse == false or (operation.DefaultResponse.HasType == false and operation.HasSuccessResponse == false)` and moved that to the two separate places where it was needed.

This duplicates some code, but in the end, I find it clearer and this allows the generated code to be cleaner.

Then, see below for the second major change